lib/hal/audio.c

```diff
@@ -103,6 +103,11 @@ void XAudioInit(int sampleSizeInBits, int numChannels, XAudioCallback callback,
 	KIRQL irql;
 	ULONG vector;
 
+	// Hack to prevent an assertion in MmGetPhysicalAddress by locking the memory.
+	// A future API redesign should use proper allocation
+	// (MmAllocateContiguousMemory) instead.
```

Comment:
I feel like this should be done as part of this PR. It's probably a trivial change because every function already uses `volatile AC97_DEVICE *pac97device;` and we can assume that nobody used the object externally. You can allocate it in this function.

I don't think we'll have a new API for AC97 anytime soon.

Reply:
I thought about doing it, but then I'd probably check for other issues, would probably have to add a shutdown-function to it etc. and that's just a bit too much for me atm. (I haven't even caught up reviewing / improving PRs yet)
(cross-post from Discord)